WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/BlogListSiteView.swift

```diff
@@ -45,40 +45,41 @@ private struct BlogListBadgeView: View {
     }
 }
 
-final class BlogListSiteViewModel: Identifiable {
-    var id: TaggedManagedObjectID<Blog> { TaggedManagedObjectID(blog) }
+struct BlogListSiteViewModel: Identifiable {
+    let id: TaggedManagedObjectID<Blog>
     let title: String
     let domain: String
     let icon: SiteIconViewModel
     let searchTags: String
-    var badge: Badge?
-
-    var siteURL: URL? {
-        blog.url.flatMap(URL.init)
-    }
+    let siteURL: URL?
+    let badge: Badge?
 
     struct Badge {
         let title: String
         let color: Color
     }
 
     func makeIcon(with size: SiteIconViewModel.Size) -> SiteIconViewModel {
-        SiteIconViewModel(blog: blog, size: size)
-    }
+        let context = ContextManager.shared.mainContext
+        guard let blog = try? context.existingObject(with: id) else { return icon }
```

Comment:
What are the advantages of fetching the existing object that the app already has in memory?

Reply:
I tried a thing to improve performance of instantiating this type, and this part is a left over from that experiment. I have reverted it back to storing `Blog` in https://github.com/wordpress-mobile/WordPress-iOS/pull/23561/commits/657bf27bd9ceefddbf024f9b7c4a6a6043d67feb.